internal/venice-client-common/src/main/java/com/linkedin/venice/compression/NoopCompressor.java

```diff
@@ -40,14 +39,8 @@ public ByteBuffer decompress(byte[] data, int offset, int length) throws IOExcep
   }
 
   @Override
-  public ByteBuffer decompressAndMaybePrependSchemaHeader(
-      byte[] data,
-      int offset,
-      int length,
-      boolean prependSchemaHeader) throws IOException {
-    if (prependSchemaHeader && offset < SCHEMA_HEADER_LENGTH) {
-      throw new VeniceException("Start offset does not have enough room for schema header.");
-    }
+  public ByteBuffer decompressAndPrependSchemaHeader(byte[] data, int offset, int length, int schemaHeader)
```

Comment:
So this function is really prepending `schemaHeader`..
Is it expected?

Reply:
As we discussed offline, we are passing the schema header retrieved into the method, so it should be OK to prepend inside the method right?